x-pack/plugins/enterprise_search/server/collectors/connectors/telemetry.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IClusterClient, Logger } from '@kbn/core/server';
+
+import { CONNECTORS_INDEX } from '@kbn/search-connectors';
+import { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+
+interface Telemetry {
+  native: {
+    total: number;
+  };
+  clients: {
+    total: number;
+  };
+}
+
+/**
+ * Register the telemetry collector
+ */
+
+export const registerTelemetryUsageCollector = (
+  usageCollection: UsageCollectionSetup,
+  client: IClusterClient,
+  log: Logger
+) => {
+  const telemetryUsageCollector = usageCollection.makeUsageCollector<Telemetry>({
+    type: 'connectors',
+    fetch: async () => fetchTelemetryMetrics(client, log),
+    isReady: () => true,
+    schema: {
+      native: {
+        total: { type: 'long' },
+      },
+      clients: {
+        total: { type: 'long' },
+      },
+    },
+  });
+  usageCollection.registerCollector(telemetryUsageCollector);
+};
+
+/**
+ * Fetch the aggregated telemetry metrics
+ */
+
+export const fetchTelemetryMetrics = async (
+  client: IClusterClient,
+  log: Logger
+): Promise<Telemetry> => {
+  const defaultTelemetryMetrics: Telemetry = {
+    native: {
+      total: 0,
+    },
+    clients: {
+      total: 0,
+    },
+  };
+
+  try {
+    const nativeCountResponse = await client.asInternalUser.count({
```

Comment:
NIT: those 2 calls could be done in parallel.

Reply:
Good to know. Could you point me to any example? Thank you

Reply:
Just something like

```ts
const [firstCall, secondCall] = await Promise.all([
  client.asInternalUser.firstAPI(..),
  client.asInternalUser.secondAPI(..),
])
```

Reply:
Any reason for not using the esClient passed into the `fetch` function directly?

Reply:
Hi @Bamieh , where is the `esClient` passed? What kind of privileges it has? Thanks.

Reply:
Hi @Bamieh , did you mean passing `coreStart.elasticsearch.client.asInternalUser` directly?

Reply:
Hi @Bamieh , I have changed it to use the passed in `esClient` and it worked.

However, I'm struggling to write the test. As `esClient` is passed implicitly, I think what I need to do is to mock the method `count` of class `ElasticsearchClient`. I tried with 
```typescript
jest.spyOn(ElasticsearchClient.prototype, 'count').mockImplementation((query: any) =>
  Promise.resolve({
    count: query.query.bool.filter[0].term.is_native ? 5 : 2,
  })
);
```

But it gave me error:
```
TS2693:  ElasticsearchClient  only refers to a type, but is being used as a value here.
```

While I'm still searching on the internet, I'd appreciate if you have a quick answer.